oak_crypto/src/encryption_key.rs

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright 2024 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use alloc::{boxed::Box, vec::Vec};
+
+use anyhow::Context;
+use async_trait::async_trait;
+use zeroize::Zeroizing;
+
+use crate::{
+    encryptor::ClientEncryptor,
+    hpke::{
+        generate_kem_key_pair, setup_base_recipient, Deserializable, PrivateKey, RecipientContext,
+        Serializable, OAK_HPKE_INFO,
+    },
+    proto::oak::crypto::v1::EncryptedRequest,
+    EMPTY_ASSOCIATED_DATA,
+};
+
+/// Generates a random encryption key pair and returns an instance of the `EncryptionKey` and a
+/// NIST P-256 SEC1 encoded point public key.
+/// <https://secg.org/sec1-v2.pdf>
+pub fn generate_encryption_key_pair() -> (EncryptionKey, Vec<u8>) {
+    let (private_key, public_key) = generate_kem_key_pair();
+    (
+        EncryptionKey::new(private_key),
+        public_key.to_bytes().to_vec(),
+    )
+}
+
+// TODO(#4513): Implement `Zeroize` for `EncryptionKey`.
```

Comment:
I thinkg the private key already implements it: https://github.com/rozbb/rust-hpke/blob/main/src/dhkex/x25519.rs#L17-L20

Reply:
Thanks

So we don't need to do this manually

Removing the comment